scripts/target.js

```diff
@@ -0,0 +1,281 @@
+    .toString(36)
+    .substring(2, 9)}-${Date.now()
+    .toString(36)}`;
+}
+
+/**
+ * Get a cookie by name.
+ * @param name
+ * @returns {string}
+ */
+function getCookie(name) {
+  const cookies = document.cookie.split(';');
+  return cookies.find((cookie) => {
+    const [key, value] = cookie.split('=');
+    return key.trim() === name && value;
+  });
+}
+
+/**
+ * Set a cookie.
+ * @param name
+ * @param value
+ * @param days
+ */
+function setCookie(name, value, days) {
+  document.cookie = `${name}=${value}; max-age=${days * 24 * 60 * 60}; path=/`;
+}
+
+/**
+ * Get the session ID.
+ * @returns {string}
+ */
+function getSessionId() {
+  const existingSessionId = getCookie(SESSION_COOKIE_NAME);
+  if (existingSessionId) {
+    return existingSessionId;
+  }
+  const newSessionId = uuid();
+  setCookie(SESSION_COOKIE_NAME, newSessionId, SESSION_COOKIE_EXPIRATION_DAYS);
+  return newSessionId;
+}
+
+/**
+ * Get all offers from a response.
+ * @param data
+ * @returns {*[]}
+ */
+function getApplicableOffers(data) {
+  const offers = [];
+  const options = data.execute?.pageLoad?.options ?? [];
+  console.debug(`received ${options.length} options`); // eslint-disable-line no-console
+  options.forEach((option) => {
+    if (option.type === 'actions') {
+      option.content.forEach((content) => {
+        console.debug('processing content', content); // eslint-disable-line no-console
+        if (['setHtml', 'insertAfter', 'insertBefore'].includes(content.type)) {
+          offers.push(content);
+        }
+      });
+    }
+  });
+  return offers;
+}
+
+/**
+ * Fetch offers for a client and a host.
+ * @param client
+ * @param sessionId
+ * @param useProxy Whether to use the proxy.
+ * @returns {Promise<any>}
+ */
+async function fetchOffers(client, sessionId, useProxy) {
+  const url = `${window.location.protocol}//${window.location.host}`;
+
+  console.debug(`Loading offers for client ${client} and url ${url}`); // eslint-disable-line no-console
+
+  const payload = {
+    context: {
+      channel: 'web',
+      address: {
+        url,
+      },
+    },
+    execute: {
+      pageLoad: {},
+    },
+  };
+
+  const options = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      'cache-control': 'no-cache',
+    },
+    body: JSON.stringify(payload),
+  };
+
+  const host = useProxy ? '/' : `https://${client}.tt.omtrdc.net/`;
+  console.debug(`Using target host: ${host}`); // eslint-disable-line no-console
+  const response = await fetch(`${host}rest/v1/delivery?client=${client}&sessionId=${sessionId}`, options);
+  if (!response.ok) {
+    throw new Error(`Failed to fetch offers: ${response.status} ${response.statusText}`);
+  }
+  const data = await response.json();
+
+  return getApplicableOffers(data);
+}
+
+/**
+ * Get the main element after it is decorated.
+ */
+function getDecoratedContent() {
+  return new Promise((resolve) => {
+    if (document.body.classList.contains('appear')) {
```

Comment:
I'm not a huge fan of having a tight coupling with the `appear` class that is `aem.js` specific. I'd probably tie this rather to a `getComputedStyle` on the `body` and check its visibility